src/index.ts

```diff
@@ -0,0 +1,65 @@
+/// <reference path="./providers/providers.d.ts" />
+import { Frame } from "tns-core-modules/ui/frame";
+import { LoadEventData } from "tns-core-modules/ui/web-view/web-view";
+import { HttpResponse } from "tns-core-modules/http/http";
+import { TnsOaProvider, TnsOaProviderType } from "./providers";
+
+export declare interface ITnsOAuthTokenResult {
+  accessToken: string;
+  refreshToken: string;
+  idToken: string;
+  accessTokenExpiration: Date;
+  refreshTokenExpiration: Date;
+  idTokenExpiration: Date;
+}
+
+export type TnsOAuthClientLoginBlock = (
+  tokenResult: ITnsOAuthTokenResult,
+  error
+) => void;
+export type TnsOAuthClientLogoutBlock = (
+  error
+) => void;
+export type TnsOAuthPageLoadStarted = (args: LoadEventData) => void;
+export type TnsOAuthPageLoadFinished = (args: LoadEventData) => void;
+
+export type TnsOAuthResponseBlock = (
+  data?: any,
+  response?: HttpResponse,
+  error?: Error
+) => void;
+
+export declare class TnsOAuthClient {
+  // private loginController;
+  provider: TnsOaProvider;
+  tokenResult: ITnsOAuthTokenResult;
+  codeVerifier?: string;
+  pkce?: boolean;
+  constructor(providerType: TnsOaProviderType, pkce?: boolean);
+  loginWithCompletion(completion?: TnsOAuthClientLoginBlock): void;
+  logoutWithCompletion(completion?: TnsOAuthResponseBlock): void;
+  refreshTokenWithCompletion(completion?: TnsOAuthClientLoginBlock): void;
+  resumeWithUrl(url: string): void;
+  logout(successPage?: string): void;
+  // private removeCookies();
```

Comment:
Maybe we can remove all the code that is commented in.

Reply:
commented code is now removed